messaging/src/main/java/org/axonframework/eventhandling/GlobalSequenceTrackingToken.java

```diff
@@ -112,6 +112,15 @@ public boolean covers(TrackingToken other) {
         return otherToken == null || otherToken.globalIndex <= this.globalIndex;
     }
 
+    @Override
+    public boolean equalsLatest(TrackingToken other) {
+        isTrue(other == null || other instanceof GlobalSequenceTrackingToken,
+                () -> "Incompatible token type provided:" + (other != null ? other.getClass().getSimpleName() : "null"));
+        GlobalSequenceTrackingToken otherToken = (GlobalSequenceTrackingToken) other;
+
+        return otherToken != null && otherToken.globalIndex == this.globalIndex;
```

Comment:
You can indeed omit this `null`-check here. Granted, the `covers` method does this too 😅